### ISSUE TITLE
Can not deserialize com.yubico.webauthn.RegisteredCredential with jackson

### ISSUE THREAD

**Issue by salehdeh76** [Yubico/java-webauthn-server#61]
I want to store CredentialRegistration of each user in database. Serializer is working as expected.
I serialize that with this code which works fine:
```
Collection<CredentialRegistration> registrations = ....;
JacksonCodecs.json().writeValueAsString(registrations)

```
It stores something like this:

```
[
    {
        "signatureCount": 0,
        "credential": {
            "credentialId": "Qh5671PvhA_YYyUE1rZ_oh45e2KiHXm7uQj6MthdfHj3D3DoHNNy1IEBbw7eDLhSRaf440gYeU-mW9xauPAXww",
            "userHandle": "lA_444PjkzZWA2PiTafgdfsuty2jKv44wedfgev8tPc",
            "publicKeyCose": "pQMmAQIsdfv-tZ3fkpkfsdfsdfsdfdfsdf36dgpTVqKiUAxIhjgvbn8jWv9bdsdfsdfsdfc_fND3WGvbnbvnzRpL3U",
            "signatureCount": 0
        },
        "userIdentity": {
            "name": "1",
            "displayName": "admin",
            "id": "lA_gWtPwerbnnvbnvbnrhtryhwetgefg2jvbnbn7ev8tPc"
        },
        "credentialNickname": "default",
        "username": "1",
        "registrationTime": "2020-03-14T23:15:51.444Z"
    }
]
```

### Deserialization error

But when I attempt to deserialize it back, using code bellow:

`JacksonCodecs.json().readValue(temp, new TypeReference<Collection<CredentialRegistration>>(){});
`

It produces this error:

> Cannot construct instance of `com.yubico.webauthn.RegisteredCredential` (no Creators, like default construct, exist): cannot deserialize from Object value (no delegate- or property-based Creator)

RegisteredCredential class is annotated using lombok but why jackson can not deserialize it?
Is there a workaround?

**Comment by emlun** [comment]
Hi! Sorry for the late response.

The `RegisteredCredential` class was intended as an abstraction rather than a ready-to-go storage type, and I expected users could use it to wrap whatever database solution they have rather than store it directly... but I guess it does make sense to support serializing it directly. It looks like deserialization fails because there's no Jackson-annotated constructor available. I'll look into supporting this in the next release, but I can't currently say when that will be.

In the meantime, one workaround is to create your own (de)serializable class with the same structure and transform between the two types. It should also be possible to register your own Jackson deserializer for a specific type, something like this: https://www.baeldung.com/jackson-deserialization .

**Comment by emlun** [comment]
This should be fixed in version `1.7.0-RC1`, which should come up on Maven Central soon. Please try it!

**Comment by emlun** [comment]
The fix is now released in version `1.7.0`. Thanks for your feedback!
